### PR TITLE
Cleanups to Interaction

### PR DESCRIPTION
**Pull request by d2dyno1**
This PR aims to reduce `IShellPage` references in `Interaction` and also moves some stuff around

Motivation:
Since `IShellPage` contains a reference to `Interaction` and `Interaction` contains `IShellPage` field it creates circular references which causes memory leaks. To solve such problem we must get rid of that reference in `Interaction` completely. Ideally, we'd have `Interaction` class completely removed and all handlers moved to respective View Models.
In my next PR I'll try to move as many functions out of `Interaction` as possible.

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@d2dyno1 Are there any noticeable improvements to memory usage with these changes?

**Comment by d2dyno1** [comment]
> @d2dyno1 Are there any noticeable improvements to memory usage with these changes?

Not really, but by going step by step this will make moving out code out of Interaction easier, and eventually fix this memory leak. @yair100

**Comment by d2dyno1** [comment]
In my next PRs I'll focus on removing as much code from `Interaction` as possible, moving functions out of this class and to ultimately remove `Interaction`. The reason I want Interaction gone, is to remove circular dependency between it and `IShellPage`. Moving out functions will also help us in the future make Files MVVM.

What are your thoughts? Help is also welcome ;)
@duke7553 @gave92 @yair100

File: Files/Interacts/Interaction.cs [hunk 1]
```diff
@@ -894,13 +830,17 @@ private async void Manager_DataRequested(DataTransferManager sender, DataRequest
                 }
                 else if (item.PrimaryItemAttribute == StorageItemTypes.Folder)
                 {
-                    await AssociatedInstance.FilesystemViewModel.GetFolderFromPathAsync(item.ItemPath)
-                        .OnSuccess(folderAsItem => items.Add(folderAsItem));
+                    if (await StorageItemHelpers.ToStorageItem<StorageFolder>(item.ItemPath) is StorageFolder folder)
```
Comment (gave92) [inline Files/Interacts/Interaction.cs:833]: I'd also pass `AssociatedInstance` to the method like this:
```cs
await StorageItemHelpers.ToStorageItem<StorageFolder>(item.ItemPath, AssociatedInstance)
```
Difference being that this should also work inside e.g phone storage. Same thing below.

Might still be posssible to get rid of the AssociatedInstance by changing ToStorageItem() somehow..

File: Files/Interacts/Interaction.cs [hunk 2]
```diff
@@ -1386,13 +1323,13 @@ private void JumpTimer_Tick(object sender, object e)
         public async Task<string> GetHashForFileAsync(ListedItem fileItem, string nameOfAlg, CancellationToken token, Microsoft.UI.Xaml.Controls.ProgressBar progress)
         {
             HashAlgorithmProvider algorithmProvider = HashAlgorithmProvider.OpenAlgorithm(nameOfAlg);
-            StorageFile itemFromPath = await AssociatedInstance.FilesystemViewModel.GetFileFromPathAsync((fileItem as ShortcutItem)?.TargetPath ?? fileItem.ItemPath);
-            if (itemFromPath == null)
+            StorageFile file = await StorageItemHelpers.ToStorageItem<StorageFile>((fileItem as ShortcutItem)?.TargetPath ?? fileItem.ItemPath);
```
Comment (gave92) [inline Files/Interacts/Interaction.cs:1326]: Same comment as above

**Comment by gave92** [comment]
@duke7553 do you have any new insight about the memory leak issue? Do you think this "down-with-the-circular-references" movement can actually help when only "normal c# objects/classes" are involved?

**Comment by duke7553** [comment]
@gave92 My position is the same as yours. While there is no evidence that these changes can improve the memory usage situation, it is still preferrable to have them for code quality purposes.

File: Files/Helpers/XamlHelpers/DependencyObjectHelpers.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+
+namespace Files.Helpers.XamlHelpers
+{
+    public static class DependencyObjectHelpers
+    {
```
Comment (duke7553) [inline Files/Helpers/XamlHelpers/DependencyObjectHelpers.cs:13]: Excellent organization here! We want contributors to find these helpers with ease.

File: Files/Interacts/Interaction.cs [hunk 4]
```diff
@@ -1329,17 +1266,17 @@ private void NewFile(ShellNewEntry itemType)
             CreateFileFromDialogResultType(AddItemType.File, itemType);
         }
 
-        public RelayCommand SelectAllContentPageItems => new RelayCommand(() => SelectAllItems());
+        public RelayCommand SelectAllContentPageItems => new RelayCommand(() => SelectAllItems(AssociatedInstance.ContentPage));
```
Comment (lukeblevins) [inline Files/Interacts/Interaction.cs:1269]: I'd suspect @yair100 has some good opinions about where we should place the RelayCommands for code quality.

**Review by lukeblevins — COMMENTED** [review]
Looks good to me once @gave92 's feedback is addressed

File: Files/Interacts/Interaction.cs (hunk 4 above)
Comment (yaichenbaum) [reply Files/Interacts/Interaction.cs:1269]: @duke7553 I would make separate classes to group relay commands depending on the kind of actions. For example, clipboard actions could go be grouped together, selection commands could be placed in a separate class etc.

**Comment by d2dyno1** [comment]
Comments addressed! @gave92 @duke7553

**Review by gave92 — APPROVED** [review]
Looks good. Thanks for this @d2dyno1

**Comment by d2dyno1** [comment]
> Looks good. Thanks for this @d2dyno1

This is just the beginning :)

**Review by lukeblevins — APPROVED** [review]

**Pull request merged**